### ISSUE TITLE
Standard exception should not require its text provided

### ISSUE THREAD

**Issue by grv87** [spdx/tools#186]
The following code:
```
LicenseInfoFactory.parseSPDXLicenseString("GPL-2.0-only WITH Classpath-exception-2.0").verify();
```
returns validation error: `Missing required license exception text`.

`Classpath-exception-2.0` exception is standard, from the list.
So, it should not require its text provided.

**Comment by goneall** [comment]
@grv87 Thanks for reporting the error.

I researched the issue and the LicenseInfoFactory currently does not support listed SPDX exceptions and is not filling in the license text.

**Comment by goneall** [comment]
To resolve this issue, I would propose the following changes:

- Change the method [getLicenseInfoFromModel](https://github.com/spdx/tools/blob/8ef7dd0b396bd5ce06f58f2c737e974792a45207/src/org/spdx/rdfparser/license/LicenseInfoFactory.java#L52) to support both exceptions licenses.  This would create a problem with the type returned since Exceptions are not a subclass of AnyLicenseInfo.  To resolve this, we could change this method to a private method returning Object and implement to public methods - one for licenses and one for exceptions.
- Add methods `LicenseInfoFactory.isSPdxListedExceptionId(String exceptionId)` and `LicenseException LicenseInfoFactory.getListedExceptionById(String exceptionId)`
- Change the [license parser WITH handling code](https://github.com/spdx/tools/blob/8ef7dd0b396bd5ce06f58f2c737e974792a45207/src/org/spdx/rdfparser/license/LicenseExpressionParser.java#L152) to check for standard exceptions.

**Comment by goneall** [comment]
Since this is more than a couple hours work, I probably won't be able to fix this for a couple weeks.  If anyone wants to create a pull request to resolve the issue, please do so - just assign the issue to yourself so we're not both working on it at the same time.

**Comment by goneall** [comment]
I started working on this issue and ran into another bug #187 .  Once this is fixed and deployed, I'll complete the fix.

The work in progress is in the branch licenseexceptions.

**Comment by goneall** [comment]
Resolved with PR #189 

@grv87 Let me know if you need a new release pushed to Maven Central

**Comment by grv87** [comment]
@goneall, thanks a lot.

No rush for release.
As a workaround I've set exception text manually before validation.
